core/services/gateway/gateway.go

```diff
@@ -0,0 +1,80 @@
+package gateway
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+)
+
+type Gateway interface {
+	job.ServiceCtx
+}
+
+type gateway struct {
+	codec    Codec
+	handlers map[string]Handler
+	connMgr  ConnectionManager
+	lggr     logger.SugaredLogger
+}
+
+func NewGatewayFromConfig(config *GatewayConfig, lggr logger.SugaredLogger) (Gateway, error) {
+	codec := &JsonRPCCodec{}
+
+	handlers := make(map[string]Handler)
+	donConnMgrs := make(map[string]DONConnectionManager)
+	for _, donConfig := range config.Dons {
+		donConfig := donConfig
+		if donConfig.DonId == "" {
+			return nil, errors.New("empty DON ID")
+		}
+		handler, err := NewHandler(donConfig.HandlerName)
```

Comment:
worst case your `switch` from NewHandler() must go here

Reply:
I'll keep it in the factory function (NewHandler) for now.